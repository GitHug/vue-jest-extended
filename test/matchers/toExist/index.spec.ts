import FakeWrapper from '../../utils';
import matcher from '../../../src/matchers/toExist';

expect.extend(matcher);

describe('.toExist', () => {
  const wrapper = new FakeWrapper();

  test('passes when wrapper exists', () => {
    wrapper.existing = true;
    expect(wrapper).toExist();
  });

  test('fails when wrapper does not exist', () => {
    wrapper.existing = false;
    expect(() => {
      expect(wrapper).toExist();
    }).toThrowErrorMatchingSnapshot();
  });

  describe('.not.toExist', () => {
    test('passes when wrapper does not exist', () => {
      wrapper.existing = false;
      expect(wrapper).not.toExist();
    });

    test('fails when wrapper exists', () => {
      wrapper.existing = true;
      expect(() => {
        expect(wrapper).not.toExist();
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
