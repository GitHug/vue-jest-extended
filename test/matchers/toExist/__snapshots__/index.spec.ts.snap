// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toExist .not.toExist fails when wrapper exists 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).not.toExist([22m[32mexpected[39m[2m)[22m


  Expected wrapper not to exist:

    [32mfalse[39m

  received:
    [31mtrue[39m
  "
`;

exports[`.toExist fails when wrapper does not exist 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toExist([22m[32mexpected[39m[2m)[22m


  Expected wrapper to exist:

    [32mtrue[39m

  received:
    [31mfalse[39m
  "
`;
